tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.Dump.cs

```diff
@@ -0,0 +1,289 @@
+// <copyright file="ProbeExpressionParser.Dump.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using Datadog.Trace.Debugger.Snapshots;
+using Datadog.Trace.Util;
+using static Datadog.Trace.Debugger.Expressions.ProbeExpressionParserHelper;
+
+namespace Datadog.Trace.Debugger.Expressions;
+
+internal partial class ProbeExpressionParser<T>
+{
+    private Expression DumpExpression(Expression expression, List<ParameterExpression> scopeMembers)
+    {
+        if (SupportedTypesService.IsSafeToCallToString(expression.Type, includeCollection: false))
+        {
+            return Expression.Call(expression, GetMethodByReflection(typeof(object), nameof(object.ToString), Type.EmptyTypes));
+        }
+
+        var stringConcat = GetMethodByReflection(typeof(string), nameof(string.Concat), new[] { typeof(object[]) });
+
+        if (IsMicrosoftException(expression.Type))
```

Comment:
Why do we need to special-case this?

Reply:
Because we a special `tostring` for it

Reply:
Ah, I see, for known Exception types we can assume it's safe to call `.Message` whereas the others might have overriden it in a way that could cause side effects. Can you please explain this in a comment?

Reply:
Fixed in e08a5f943cbd55627c65966e5e2a4a5e4674e166